staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -223,3 +226,28 @@ func (f *FieldManager) buildManagerInfo(prefix string, operation metav1.ManagedF
 	}
 	return internal.BuildManagerIdentifier(&managerInfo)
 }
+
+// stripSet is the list of fields that should never be part of a mangedFields.
+var stripSet = fieldpath.NewSet(
+	fieldpath.MakePathOrDie("apiVersion"),
+	fieldpath.MakePathOrDie("kind"),
+	fieldpath.MakePathOrDie("metadata", "name"),
+	fieldpath.MakePathOrDie("metadata", "namespace"),
+	fieldpath.MakePathOrDie("metadata", "creationTimestamp"),
+	fieldpath.MakePathOrDie("metadata", "selfLink"),
+	fieldpath.MakePathOrDie("metadata", "uid"),
+	fieldpath.MakePathOrDie("metadata", "resourceVersion"),
+)
+
+// stripFields removes a predefined set of paths found in typed from managed and returns the updated ManagedFields
+func (f *FieldManager) stripFields(managed fieldpath.ManagedFields, manager string) fieldpath.ManagedFields {
+	vs, ok := managed[manager]
```

Comment:
I think it's OK. It probably can't happen that we have `manager` in the map but with a `nil` pointer (in which case you would have both `ok == true` and `vs == nil` at the same time).
@jennybuckley ?

Reply:
Right, as far as I saw we always only delete the manager entirely. Couldn't find it being set to nil.
Could happen if somebody changes this in the future and doesn't know, so maybe a check would be safer? But I really wouldn't expect it 🤔

Reply:
Yeah, hopefully this would fail in some ways before going anywhere, but still might be better to check ...

Reply:
But I wouldn't make it fail (I think I would want to, but just adding an error for this seems overkill).
Will add this tomorrow I guess.

Reply:
yeah, it's definitely not an error, it's just something that needs to be removed from the list?

Reply:
So I updated again. But i just leave the managers untouched when i find a nil one.
This is on purpose as I then would expect other tests to fail if it happens rather than silently solving another developers "mistake" once it happens.

But I guess another test for this would be good :thinking: if not already covered by the existing integration tests.